apps/src/code-studio/progressReduxSelectors.js

```diff
@@ -117,6 +123,25 @@ export const getLevelPropertiesPath = state => {
   }
 };
 
+/**
+ * Get the ID of the current level, or sublevel if we're on a sublevel.
+ */
+export const getCurrentLevelOrSublevelId = state => {
```

Comment:
This is the main thing I'm stumped on; currently, `state.progress.currentLevelId` refers to the current non-sublevel ID. So if we're on a sublevel of a bubble choice level, that ID will refer to the parent bubble choice level and not the sublevel. Initially I thought it would make sense to update `state.progress.currentLevelId` to the current level or sublevel ID, but we use this field in a bunch of other places so I'm wary of breaking existing behavior. For Lab2, we need the level ID of the specific sublevel we're on in order to fetch the right level properties, etc, so we need some way to get that level's ID (which is what this selector does). It doesn't feel great, because it's ambiguous what `currentLevelId` even means then, but I thought this might be safer for now. Curious what folks think (cc @molly-moen @breville)

Reply:
How much do the functionality of a level and a sublevel overlap? Mostly I'm wondering if we provide a sublevel id to the apis that we are currently using for "regular" levels, will any break? Do sublevel queries always happen with the parent level and sublevel ids or do they just use the sublevel?

Reply:
yeah I was wondering if we should add a `parentLevelId` field in redux that is null if this isn't a sublevel, and has the parent id otherwise.

Reply:
Oh yeah, that's a good idea. Maybe the current script level ID (which would be the parent level ID) in addition to the current level ID (which would be the sublevel's ID if on a sublevel, or parent level's level ID)